### PR TITLE
[ELY-1850] CredentialStore API missing ability to check alias types

### PR DESCRIPTION
**Pull request by lvydra**
https://issues.redhat.com/browse/ELY-1850

### CODE REVIEW

**Comment by Skyllarr** [comment]
@lvydra Thank you for PR! Can you please add a test for this as well?

File: credential/store/src/main/java/org/wildfly/security/credential/store/CredentialStore.java [hunk 1]
```diff
@@ -354,6 +354,17 @@ public void remove(String credentialAlias, Class<? extends Credential> credentia
...
     }
 
+    /**
+     * Returns {@code Set<String>} of types stored in this store with given alias.
+     *
+     * @param credentialAlias to find the credentials types in the store
+     * @return {@code Set<String>} of all credential types stored in this store with given alias
+     * @throws UnsupportedOperationException when this method is not supported by the underlying credential store
+     */
+    public Set<String> getStoredTypesForAlias(String credentialAlias) throws UnsupportedOperationException {
```
Comment (fjuma) [inline credential/store/src/main/java/org/wildfly/security/credential/store/CredentialStore.java:364]: Just a small comment, to be consistent with the parameter name used when removing a credential, perhaps we could rename this method to `getCredentialTypesForAlias`.

File: credential/store/src/main/java/org/wildfly/security/credential/store/CredentialStoreSpi.java [hunk 2]
```diff
@@ -148,6 +148,19 @@ public void flush() throws CredentialStoreException {
...
+    /**
+     * Returns credential types stored in this store with given alias as {@code Set<String>}.
+     *
+     * It is not mandatory to override this method (throws {@link UnsupportedOperationException} by default).
+     *
+     * @param credentialAlias to find the credentials types in the store
+     * @return {@code Set<String>} of all credential types stored in this store with credential alias
+     * @throws UnsupportedOperationException when this method is not supported by the underlying credential store
+     */
+    public Set<String> getStoredTypesForAlias(String credentialAlias) throws UnsupportedOperationException {
```
Comment (fjuma) [inline credential/store/src/main/java/org/wildfly/security/credential/store/CredentialStoreSpi.java:160]: Same here.

File: tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java [hunk 3]
```diff
@@ -97,6 +97,7 @@
     public static final String ALIAS_ARGUMENT = "alias";
     public static final String CHECK_ALIAS_PARAM = "exists";
     public static final String ALIASES_PARAM = "aliases";
+    public static final String STORED_TYPES = "stored-types";
```
Comment (fjuma) [inline tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java:100]: `CREDENTIAL_TYPES="credential-types"`

File: tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java [hunk 4]
```diff
@@ -566,6 +571,21 @@ private void aliases(CredentialStore credentialStore) throws Exception {
         setStatus(ElytronTool.ElytronToolExitStatus_OK);
     }
 
+    private void aliasStoredTypes(CredentialStore credentialStore) {
```
Comment (fjuma) [inline tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java:574]: Same here.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks @lvydra! Just have some small comments.

**Comment by lvydra** [comment]
@fjuma Thanks, fixed.

**Review by fjuma — APPROVED** [review]
Thanks @lvydra!

**Pull request merged**